R/beers-r6.R

```diff
@@ -0,0 +1,153 @@
+Beers_Interpolator <- R6::R6Class("Beers_Interpolator",
+
+  public = list(
+    b1 = NULL,
+    b2 = NULL,
+    bm = NULL,
+    bp = NULL,
+    bf = NULL,
+
+    initialize = function(b1, b2, bm, bp, bf) {
+      self$b1 <- b1
+      self$b2 <- b2
+      self$bm <- bm
+      self$bp <- bp
+      self$bf <- bf
+    },
+
+    interpolate = function(points) {
+      if (length(points) < 6) {
+        stop("Beers interpolation requires at least 6 points")
+
+      } else if (!is.numeric(points)) {
+        stop("Beers interpolation requires numeric argument")
+      }
+
+      res <- rep(0, ((length(points) * 5) - 4))
+      for (p in 1:(length(points) - 1)) {
+
+        if (p == 1) {
+          for (i in 1:5) res[i] <-
+            sum(points[1:6] * self$b1[i, 1:6])
+
+        } else if (p == 2) {
+          for (i in 1:5) res[i + 5] <-
+            sum(points[1:6] * self$b2[i, 1:6])
+
+        } else if (p == length(points) - 2) {
+          for (i in 1:5) res[(((p - 1) * 5) + i)] <-
+            sum(points[((p - 4) + 1):((p - 4) + 6)] * self$bp[i, 1:6])
+
+        } else if (p == length(points)-1) {
+          for (i in 1:6) res[(((p - 1) * 5) + i)] <-
+            sum(points[((p - 5) + 1):((p - 5) + 6)] * self$bf[i, 1:6])
+
+        } else {
+          for (i in 1:5) res[(((p - 1) * 5) + i)] <-
+            sum(points[((p - 3) + 1):((p - 3) + 6)] * self$bm[i, 1:6])
+        }
+      }
+      res
+    }
+  )
+)
+
+beers_modified_r6 <- Beers_Interpolator$new(
+  b1 = matrix(nrow = 5, ncol = 6, byrow = TRUE, data = c(
```

Comment:
Given these are package constants, I wonder if there's a better way of doing this.  I would suggest looking at sysdata.rda - this is somewhere in both R-exts and Hadley's book.

Write a script (and add that to .Rbuildignore) that duplicates the code below, but with each matrix saved as a SHOUTY_CASE name (e.g., `BEERS_COEF_1 <- matrix(...)`) and have that code save out the sysdata.rda file (see https://github.com/richfitz/ids/blob/master/bootstrap.R for an example).

Reply:
Done. I've put it in R/sysdata.rda, rather than going in data/ - when I tried the latter, it wanted me to document each matrix:-

checking for missing documentation entries ... WARNING
Undocumented code objects:
  'BEERS_INT_MOD_FINAL' 'BEERS_INT_MOD_FIRST' 'BEERS_INT_MOD_MID'
...
Undocumented data sets:
  'BEERS_INT_MOD_FINAL' 'BEERS_INT_MOD_FIRST' 'BEERS_INT_MOD_MID'
...

I guess that's not quite what data/ is for...